### PR TITLE
[#2902] Passwords are no longer leaked while running in debug mode (main)

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

File: lib/core/src/packStruct.cpp [hunk 1]
```diff
@@ -3362,7 +3363,12 @@ int packStruct(const void *inStruct,
         /* add a NULL termination */
         *static_cast<char*>(outPtr) = '\0';
         if ( getRodsLogLevel() >= LOG_DEBUG9 ) {
-            printf( "packed XML: \n%s\n", ( char * ) packedOutput.bBuf.buf );
+            const auto& bbuf = packedOutput.bBuf;
+            const std::string_view text(static_cast<char*>(bbuf.buf), bbuf.len);
+
+            if (text.find("<authPlugReqInp_PI>") == std::string_view::npos) {
```
Comment (trel) [inline lib/core/src/packStruct.cpp:3369]: same as 4-2-stable, please add comment x5

**Review by trel — APPROVED** [review]

**Review by alanking — APPROVED** [review]

File: plugins/network/src/tcp.cpp [hunk 2]
```diff
@@ -444,8 +446,14 @@ irods::error read_bytes_buf(
     _buffer->len = bytes_read;
 
     // log transaction if requested
-    if ( _protocol == XML_PROT ) {
-        rodsLog(LOG_DEBUG8, "received msg: \n%.*s\n", _buffer->len, ( char* )_buffer->buf );
+    if (_protocol == XML_PROT) {
+        const std::string_view text(static_cast<char*>(_buffer->buf), _buffer->len);
+
+        // Do not print authentication request input when using the XML protocol
```
Comment (alanking) [inline plugins/network/src/tcp.cpp:452]: Looks like there's some trailing space at the end of this line, which is probably why clang-format is upset...

**Comment by alanking** [comment]
Let's # it

**Comment by korydraughn** [comment]
Will do. Just verifying that the changes actually fix the problem.

Will also run through the testing environment to make sure things are still happy.

**Review by alanking — APPROVED** [review]
Seems fine to me. Hadn't considered the `\since` tag as it relates to 4.3.0 vs 4.2.12 (considering 4.3.0 will likely come out before 4.2.12).

**Comment by korydraughn** [comment]
I feel the approach is correct, but willing to be convinced to change `\since` to something else.

**Comment by alanking** [comment]
I think it's the right way to go, yeah.

**Comment by korydraughn** [comment]
Core tests pass.

Added #

**Pull request merged**